pkg/cri/server/helpers.go

```diff
@@ -242,6 +242,29 @@ func (c *criService) ensureImageExists(ctx context.Context, ref string, config *
 	return &newImage, nil
 }
 
+// validateTargetContainer checks that a container is a valid
+// target for a container using PID NamespaceMode_TARGET.
+// The target container must be in the same sandbox and must be running.
+// Returns the target container for convenience.
+func (c *criService) validateTargetContainer(sandboxID string, nsOpts *runtime.NamespaceOption) (containerstore.Container, error) {
+	targetContainerID := nsOpts.TargetId
+	targetContainer, err := c.containerStore.Get(targetContainerID)
+	if err != nil {
+		return containerstore.Container{}, errors.Wrapf(err, "container %q does not exist", targetContainerID)
+	}
+
+	if targetContainer.Metadata.SandboxID != sandboxID {
+		return containerstore.Container{}, errors.Errorf("container %q belongs to a different sandbox", targetContainerID)
+	}
+
+	status := targetContainer.Status.Get()
+	if state := status.State(); state != runtime.ContainerState_CONTAINER_RUNNING {
+		return containerstore.Container{}, errors.Errorf("container %q is in state %s", targetContainerID, state)
```

Comment:
`"container %q is in state %s"` -> `"container %q is not running - in state %s"`